src/blr/cli/filterclusters.py

```diff
@@ -1,367 +1,306 @@
-    parser.add_argument("-w", "--window", metavar='<INTEGER>', type=int, default=30000, help="Window size cutoff for maximum distance "
-                                                                              "in between two reads in one molecule. "
-                                                                              "DEFAULT: 30000")
-    parser.add_argument("-bc", "--barcode_tag", metavar='<STRING>', type=str, default='BC', help="Bam file tag where barcode is stored. DEFAULT: BC")
-    parser.add_argument("-f", "--filter_bam", metavar='<OUTPUT>', type=str, default=False, help="Write an output bam file where reads have their barcode tag removed if it is "
-                                                                            "from a cluster with more molecules than -M (--Max_molecules) molecules. DEFAULT: False")
-    parser.add_argument("-M", "--Max_molecules", metavar='<INTEGER>', type=int, default=500, help="When using -f (--filter) this will remove barcode tags for those clusters which have more than -M molecules. DEFAULT: 500")
-    parser.add_argument("-s", "--split", action="store_true", help="Will intead of writing one output filtered file "
-                                                                   "split output into separate files based on "
-                                                                   "#mol/bc. The -f (--filter) output file name will instead be "
-                                                                   "used as prefix for the output. -f option "
-                                                                   "required. DEFAULT: None")
+    parser.add_argument("x2_bam", help=".bam file tagged with BC:Z:<int> tags. Needs to be indexed, sorted & have "
+                                       "duplicates removed.")
+    parser.add_argument("output", help="Output filtered file.")
+
+    parser.add_argument("-t","--threshold", metavar="<INTEGER>", type=int, default=4,
+                        help="Threshold for how many reads are required for including given molecule in statistics "
+                             "(except_reads_per_molecule). DEFAULT: 4")
+    parser.add_argument("-w", "--window", metavar="<INTEGER>", type=int, default=30000,
+                        help="Window size cutoff for maximum distance in between two reads in one molecule. DEFAULT: "
+                             "30000")
+    parser.add_argument("-bc", "--barcode_tag", metavar="<STRING>", type=str, default="BC",
+                        help="Bam file tag where barcode is stored. DEFAULT: BC")
+    parser.add_argument("-s", "--stats_file", metavar="<PREFIX>", type=str,
+                        help="Write barcode/molecule statistics files. DEFAULT: None")
+    parser.add_argument("-M", "--Max_molecules", metavar="<INTEGER>", type=int, default=500,
```

Comment:
In this case I believe the long-form of the option, i.e. `--Max_molecules` , should be written in lowercase letter even though the short-form option is `-M`. This is because it will be stored under the varible name `args.Max_molecules`which is not according to PEP8 convention.

Reply:
Doesn't sound unreasonable but where is it written? Just me being curious. 

Either way, I'll change it.

Reply:
```suggestion
    parser.add_argument("-M", "--max_molecules", metavar="<INTEGER>", type=int, default=500,
```

Reply:
> Doesn't sound unreasonable but where is it written? Just me being curious.
> 
> Either way, I'll change it.

I am looking at this:
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Reply:
Ahhhh, yes of course >.<